tests/std/tests/P0896R4_ranges_alg_generate/test.cpp

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <algorithm>
+#include <cassert>
+#include <concepts>
+#include <range_algorithm_support.hpp>
+#include <ranges>
+#include <utility>
+
+constexpr void smoke_test() {
+    using ranges::generate, ranges::iterator_t;
+    using std::same_as;
+
+    int output1[] = {13, 42, 1367};
+    int output2[] = {13, 42, 1367};

```

Comment:
Again, Casey [commented](https://github.com/microsoft/STL/pull/904#discussion_r441925510) similarly on the `fill`/`fill_n` PR but since `output1` is only used in the first code block and `output2` only used in the second, can these be moved into more local scope?